cmd/thanos/sidecar.go

```diff
@@ -229,6 +229,16 @@ func runSidecar(
 		})
 	}
 
+	err := addShipper(logger, reg, g, metadata.SidecarSource, m, dataDir, objStoreConfig, uploadCompacted)
+	if err != nil {
+		return errors.Wrap(err, "failed to add block shipper to run group")
+	}
+
+	level.Info(logger).Log("msg", "starting sidecar", "peer", peer.Name())
+	return nil
+}
+
+func addShipper(logger log.Logger, reg prometheus.Registerer, g *run.Group, sourceType metadata.SourceType, m *promMetadata, dataDir string, objStoreConfig *pathOrContent, uploadCompacted bool) error {
```

Comment:
What about rule.go shipper?

Reply:
Not sure exactly what you mean here.

Reply:
Ah, I need to look at that, I'll take care of it tomorrow!

Reply:
I looked at it a bit, and I feel like the best decision around this for now is to not generalize this in the `addShipper` function, but once gossip removal lands, this is nicely generalizable for receive/sidecar/rule.

Reply:
Actually reducing this to what's necessary in receive, I think I might even prefer the slight duplication, as the code necessary is _a lot_ smaller than the `addShipper` function.